### PR TITLE
Add Overview doc for Attestation Library

### PR DESCRIPTION
**Pull request by acamadeo**
This is the first of three sections of documentation:
1) An overview of the attestation library defining its components and providing a high-level understanding of the interface.

The next two planned sections are:
2) Steps on how to use the interface (with code snippets). Will contain PGP and PKIX/JWT-specific details.
3) Actual code examples.

### CODE REVIEW

File: pkg/kritis/cryptolib/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Crypto Library
+
+The Crypto Library provides an interface for creating and verifying Attestations through a Signer and Verifier, respectively.
```
Comment (ooq) [inline pkg/kritis/cryptolib/README.md:3]: a Signer and a Verifier

File: pkg/kritis/cryptolib/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+## Components
+
+### Attestation
+
+An Attestation is a statement that a container image meets a policy and can be deployed to a Kubernetes cluster. It contains a signature generated by the private key of a trusted entity (e.g. a build system, vulnerability scanner, etc.). However, as anybody can create an Attestation, it cannot be trusted until it is verified using the entity's corresponding public key. An Attestation is defined independently from the wire format of a signature, allowing the user to store signatures generated by PGP or PKIX keys, or signatures encoded as JWTs.
+
+An Attestation contains a payload and a signature generated over the payload with the trusted entity’s private key. It also contains the ID of the public key which can verify the Attestation’s signature.
+
+### Public Key
```
Comment (ooq) [inline pkg/kritis/cryptolib/README.md:13]: We should consider either 1) inline the spec for the data structures here, or 2) link to the spec in the code.

File: pkg/kritis/cryptolib/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+### Payload
+The payload is a message provided by the trusted entity. It is signed by their private key to create a signature. By convention, the payload is a JSON-encoded string conforming to the [Red Hat Atomic Host signature format](https://github.com/aweiteka/image/blob/e5a20d98fe698732df2b142846d007b45873627f/docs/signature.md).
+
+## Interface
+
+### Signing
+
+#### Signer
+A trusted entity will use a Signer to generate Attestations. There are three signer implementations, one for each KeyType: a `PgpSigner`, `PkixSigner`, and a `JwtSigner`. Each signer has its own constructor (e.g. `NewPgpSigner`), which creates a Signer storing the trusted entity’s private key and any other data required to create an Attestation.
```
Comment (ooq) [inline pkg/kritis/cryptolib/README.md:29]: Maybe some example code?

File: pkg/kritis/cryptolib/README.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+Each signer implements the Signer interface: the `CreateAttestation` method. When passed a payload to sign over, `CreateAttestation` will generate and return an Attestation containing the signature.
+
+### Verifying
+
+#### PublicKey
+To create a PublicKey, the user can call `NewPublicKey`, passing in the public key material, key ID, KeyType, and any other data necessary to verify an Attestation.
```
Comment (ooq) [inline pkg/kritis/cryptolib/README.md:36]: Replacing the text with code will be clearer?

File: pkg/kritis/cryptolib/README.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
+# Crypto Library
```
Comment (alexcope) [inline pkg/kritis/cryptolib/README.md:1]: Attestation Library (here and elsewhere)

File: pkg/kritis/cryptolib/README.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+The Crypto Library provides an interface for creating and verifying Attestations through a Signer and Verifier, respectively.
+
+## Components
+
+### Attestation
+
+An Attestation is a statement that a container image meets a policy and can be deployed to a Kubernetes cluster. It contains a signature generated by the private key of a trusted entity (e.g. a build system, vulnerability scanner, etc.). However, as anybody can create an Attestation, it cannot be trusted until it is verified using the entity's corresponding public key. An Attestation is defined independently from the wire format of a signature, allowing the user to store signatures generated by PGP or PKIX keys, or signatures encoded as JWTs.
```
Comment (alexcope) [inline pkg/kritis/cryptolib/README.md:9]: Nit: I would phrase it like this:
An Attestation is a signed statement about a container image in a known format. A container is allowed to be deployed to a kubernetes cluster if it presents Attestations that satisfy the cluster's policy.

File: pkg/kritis/cryptolib/README.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+### Public Key
+A PublicKey is the definitive trust anchor used to verify that an Attestation’s signature is valid. Unlike Attestations, which are considered untrustworthy until verified, PublicKeys are assumed to contain trustworthy information. Consequently, this information should be provided directly by the trusted party.
+
+A PublicKey contains the raw public key material and an ID. It also contains a KeyType, one of {`Pgp`, `Pkix`, or `Jwt`}, indicating how the trusted entity stores data within the Attestation. It also contains a SignatureAlgorithm, indicating the cryptographic algorithm, padding algorithm, and hash function used on the payload to create the signature in the Attestation.
+
+### Private Key
+The trusted entity has a private key, which is used by the Signer to generate an Attestation’s signature.
+
+### Payload
+The payload is a message provided by the trusted entity. It is signed by their private key to create a signature. By convention, the payload is a JSON-encoded string conforming to the [Red Hat Atomic Host signature format](https://github.com/aweiteka/image/blob/e5a20d98fe698732df2b142846d007b45873627f/docs/signature.md).
```
Comment (alexcope) [inline pkg/kritis/cryptolib/README.md:22]: Payloads are also used by the verifier.  Add something aboutn this and how they are untrusted until the signature is verified by a trusted public key

File: pkg/kritis/cryptolib/README.md (hunk 1 above)
Comment (acamadeo) [reply pkg/kritis/cryptolib/README.md]: Done.

File: pkg/kritis/cryptolib/README.md (hunk 2 above)
Comment (acamadeo) [reply pkg/kritis/cryptolib/README.md]: Added a link to the spec.

File: pkg/kritis/cryptolib/README.md (hunk 4 above)
Comment (acamadeo) [reply pkg/kritis/cryptolib/README.md]: We plan on putting code snippets in the tutorial doc (see #569)

File: pkg/kritis/cryptolib/README.md (hunk 5 above)
Comment (acamadeo) [reply pkg/kritis/cryptolib/README.md]: Done.

File: pkg/kritis/cryptolib/README.md (hunk 6 above)
Comment (acamadeo) [reply pkg/kritis/cryptolib/README.md]: Done.

File: pkg/kritis/cryptolib/README.md (hunk 7 above)
Comment (acamadeo) [reply pkg/kritis/cryptolib/README.md]: Done.

File: pkg/kritis/cryptolib/README.md (hunk 3 above)
Comment (acamadeo) [reply pkg/kritis/cryptolib/README.md]: We plan on putting code snippets in the tutorial doc (see #569)

**Pull request merged**